File: Assignment 2/Q1a.py
size = int(input("Number of linear equidistant points:"))
points = []

for n in range(size):
	points.append(n)
	n += 1

tot_dist = 0
tot_pairs = size*size

for i in points:
	for j in points:
		tot_dist += abs(j - i)

avg_dist = tot_dist/tot_pairs

print("The average distance between two points is ", avg_dist)
'''
Number of linear equidistant points:6
The average distance between two points is  1.9444444444444444
'''

File: Assignment 2/Q1a.c
#include <stdio.h>
#include<stdlib.h>

int main(){
	
	int size;
	printf("Enter the number of linear equidistant points:");
	scanf("%d", &size);

	float tot_dist = 0;
	int tot_pairs = size*size;

	for (int i = 0; i < size; i++){
		for (int j = 0; j < size; j++){
			tot_dist += abs(j - i);
		}
	}

	float avg_dist = tot_dist/tot_pairs;

	printf("The average distance between two points is %.4f", avg_dist);

	return 0;
}
/*
Output:
Enter the number of linear equidistant points:6
The average distance between two points is 1.9444
*/